applications/plugins/SofaSphFluid/src/SofaSphFluid/OglFluidModel.inl

```diff
@@ -584,12 +584,8 @@ void OglFluidModel<DataTypes>::drawSprites(const core::visual::VisualParams* vpa
 }
 
 template<class DataTypes>
-void OglFluidModel<DataTypes>::drawVisual(const core::visual::VisualParams* vparams)
+void OglFluidModel<DataTypes>::doDrawVisual(const core::visual::VisualParams* vparams)
 {
-    if (!vparams->displayFlags().getShowVisualModels()) return;
-
-    vparams->drawTool()->saveLastState();
```

Comment:
You forgot to remove `vparams->drawTool()->restoreLastState();` at the end of the function